src/types/list.rs

```diff
@@ -83,6 +84,23 @@ impl PyList {
         }
     }
 
+    /// Gets the list item at the specified index.
+    pub fn get(&self, index: usize) -> PyResult<&PyAny> {
+        unsafe {
+            let item = ffi::PyList_GetItem(self.as_ptr(), index as Py_ssize_t);
+            self.py().from_borrowed_ptr_or_err(item)
+        }
+    }
+
+    /// Gets the list item at the specified index. Undefined behavior on bad index. Use with caution.
+    #[cfg(not(any(Py_LIMITED_API, PyPy)))]
+    pub fn get_unchecked(&self, index: usize) -> &PyAny {
```

Comment:
This one is definitely `unsafe`:

```suggestion
    /// Gets the list item at the specified index. Undefined behavior on bad index. Use with caution.
    #[cfg(not(any(Py_LIMITED_API, PyPy)))]
    pub unsafe fn get_unchecked(&self, index: usize) -> &PyAny {
```